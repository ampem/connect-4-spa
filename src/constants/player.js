export const RED_PLAYER = 'RED_PLAYER';
export const YELLOW_PLAYER = 'YELLOW_PLAYER';