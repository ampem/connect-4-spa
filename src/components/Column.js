import React from 'react';
import PropTypes from 'prop-types';
import { RED_PLAYER, YELLOW_PLAYER } from '../constants/player';
import Cell from './Cell';
import '../styles/column.css';


function Column({cells=[], clickHandler}) {
    return (
        <div className="column" onClick={clickHandler}>
            { cells.map((occupier,i)=><Cell key={i} occupiedBy={occupier} />)}
        </div>
    );
}

Column.propTypes = {
	cells: PropTypes.arrayOf(PropTypes.oneOf([RED_PLAYER, YELLOW_PLAYER])),
	clickHandler: PropTypes.func.isRequired
}

export default Column;
