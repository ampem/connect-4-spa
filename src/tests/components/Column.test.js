import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { RED_PLAYER, YELLOW_PLAYER } from '../../constants/player';
import Column from '../../components/Column';

describe('Column', () => {
	it('renders correctly', () => {
		const wrapper = shallow(
			<Column
				cells={[RED_PLAYER, YELLOW_PLAYER]}
				clickHandler={jest.fn()}
			/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('Renders empty columns correctly', () => {
		const wrapper = shallow(
			<Column
				cells={[]}
				clickHandler={jest.fn()}
			/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('Renders correctly if no cells are passed', () => {
		const wrapper = shallow(
			<Column
				clickHandler={jest.fn()}
			/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('Calls `clickHandler` when the column is clicked', () => {
		const mockClickHandler = jest.fn(x=>42);
		const wrapper = shallow(
			<Column
				clickHandler={mockClickHandler}
			/>);
		wrapper.simulate('click')
		expect(wrapper).toMatchSnapshot();
		expect(mockClickHandler.mock.calls.length).toBe(1);
	});
})
