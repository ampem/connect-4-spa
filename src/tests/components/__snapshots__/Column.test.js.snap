// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column Renders correctly if no cells are passed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    clickHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "column",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "column",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Column Renders empty columns correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    cells={Array []}
    clickHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "column",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "column",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Column renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    cells={
        Array [
            "RED_PLAYER",
            "YELLOW_PLAYER",
          ]
    }
    clickHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Cell
          occupiedBy="RED_PLAYER"
/>,
        <Cell
          occupiedBy="YELLOW_PLAYER"
/>,
      ],
      "className": "column",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "occupiedBy": "RED_PLAYER",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "occupiedBy": "YELLOW_PLAYER",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Cell
            occupiedBy="RED_PLAYER"
/>,
          <Cell
            occupiedBy="YELLOW_PLAYER"
/>,
        ],
        "className": "column",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "occupiedBy": "RED_PLAYER",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "occupiedBy": "YELLOW_PLAYER",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
